Nu stiu de ce nu imi functioneaza afisarea.

cristicrisan07 opened the pull request.

cristicrisan07 committed:
Nu stiu de ce nu imi functioneaza afisarea.

Alex-Cosma commented:
Ce anume nu funcționează?

cristicrisan07 commented:
Cand ajunge sa execute functia de  afisare , si executa printf, nu se mai intampla nimic

Alex-Cosma commented:
Ai debuguit?

cristicrisan07 commented:
Am pus in acel for din funcția de afisare un printf("da") inainte de printul mare, si am văzut ca se blochează la cel mare

Alex-Cosma commented on Crisan Cristian Alexandru/010 3.3/student.c:
```diff
@@ -0,0 +1,72 @@
+        initializeStudent(&students[i]);
+        printf("Reading student %d\n", i+1);
+        printf("Student[%d] name=",i+1);
+        scanf("%s",&students[i].name);
```
Aici e problema. Grijă mare!
`students[i].name` e deja un `char *` (adică un pointer)
nu are niciun sens să mai pui `&` înainte!
Probabil știai asta dar ai pus-o din instinct/neatenție.

Alex-Cosma replied:
Același lucru și mai jos, la adresă

cristicrisan07 committed:
Salut! În fișierul "rws.c", la linia 50 am o problemă: nu pot accesa nicicum actors[*nrOfDactors].name.
Nu merge nici strcpy, nici să afișez nu pot. Am încercat să îl afișez mai sus, dar mă lovesc de aceeași eroare.
Și ce e ciudat e că mai sus am pe aceeași structură: movies[nrOfDMovies].name, însă în acel caz nu mă lovesc de nicio eroare. Nu înțeleg ce nu am făcut bine.
Scuze de deranj dar sper că mă poți ajuta.
Sărbători fericite!

Alex-Cosma commented on Crisan Cristian Alexandru/Casualwood/rws.c:
```diff
@@ -0,0 +1,71 @@
+            printf("da5");
+            if(isUnique(actors,*nrOfDactors,info)==true)
+            {
+                strcpy(actors[*nrOfDactors].name,info);
```
Problema ta nu e aici

Alex-Cosma commented on Crisan Cristian Alexandru/Casualwood/rws.c:
```diff
@@ -0,0 +1,71 @@
+        movies[*nrOfDMovies].CastSize=0;
+        char *info=strtok(currentLine,",");
+        printf("da2");
+        strcpy(movies[*nrOfDMovies].name,info);
```
Problema ta e de fapt, aici - dacă ai fi mers cu debuggerul, ai fi văzut că de fapt aici crapă, nu mai jos.
Motivul e simplu - tu ai un array de filme, `movies`. Fiecare `movie` are un `char * name` pe lângă alte cele. Ideea e că, degeaba în funcția main ai declarat tu:
`movie *movies=(movie*)malloc(sizeof(movie)*30);`
că tu ai alocat spațiu doar pentru 30 de pointeri către 30 de obiecte de tipul `movie`, pe când, fiecare dintre aceste 30 de obiecte conțin un pointer către un `char * name` <-- *ȘI ĂSTA* trebuie alocat!

Uită-te la exemplele pe care le-am făcut eu. De fiecare dată când aloc o structură care mai are un șir de .. ceva, orice, aloc și acel șir! (de obicei am denumit funcțiile de alocare `initialize ... `

cristicrisan07 committed:
Acceea era problema, ai avut dreptate, mulțumesc!
Dar acum am o altă problemă. Am pus în funcția de citire o condiție ca în momentul în care citește actorul de la finalul unei linii, și deci intalnește caracterul special ce marchează newline, să nu îl includă și pe acel caracter în numele actorului, însă dintr-un oarecare motiv nu îmi execută acele instrucțiuni. Și nu înțeleg de ce...

Alex-Cosma commented:
Mă bucur că ai rezolvat
